elastic/kibana
# [Visualize] First version of by-value visualize editor

## What is this?
This PR implements editing visualization by value inside dashboard and returning the edited visualization back to dashboard. 

The idea is as follows:
1. We take the embeddable input from the dashboard and pass it (along with the original embeddable id) to the visualize editor via state-transfer service
2. Visualize editor reads the input from the state transfer service and renders the visualize embeddable (+ editor controls) from the input
3. Once the changes are applied, visualize editor passes the updated input together with the original embeddable id back to the dashboard
4. Dashboard updates the existing embeddable from the input it received

Note: Sharing of the visualization by-value doesn't work yet.

##  Implementation details
To support this flow, I introduced a new `edit_by_value` URL and a new `visualize_byValue_editor`. When editing visualization by value, there is no need to sync state from the URL, so the flow becomes a bit simpler. Common functionality was extracted to common components/functions, but some code duplication is probably still present.

## What didn't change?
There should be no changes to the existing visualize editor. If something is wrong with editing visualizations from saved objects, it's likely a bug not a feature :)

## How to test this?
1. The main focus should be on testing the existing visualize editor and that nothing is broken. 

2. To test the new functionality, you'll need to set the `showNewVisualizeFlow` config value to `true`:
https://github.com/elastic/kibana/blob/f5061c7c5139da7de62e8ede2e12fe38b45cab2f/src/plugins/visualize/config.ts#L23
This requires a server restart!

3. Once the flag has been enabled:
  a) create a dashboard
  b) create a new visualization (any visualization except Lens) and add it to dashboard
  c) from the panel action menu select "Edit visualization" -> this should trigger the new "by-value" editor 

## Where are functional/unit tests?
Incoming!

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
